### DISCUSSION TITLE
When requesting a CNAME record with an HTTPS (type 65) DNS query is it normal that Adguard doesnt resolve the whole chain?

### DISCUSSION THREAD

**Discussion by Buco7854** [AdguardTeam/AdGuardHome#7180] [Q&A]
I have 2 DNS rewrites.

reverse-proxy-server-1.lan -> 192.168.1.61
*.mydomain.com -> reverse-proxy-server-1.lan

If a `A` query is made to `test.mydomain.com` then Adguard answer with

`test.mydomain.com` -> `reverse-proxy-server-1.lan`
`reverse-proxy-server-1.lan` -> `192.168.1.61`

But if an `HTTPS` (type 65) query is made then adguard answers with
`test.mydomain.com` -> `reverse-proxy-server-1.lan`
(In this case the Authorative Nameservers field is also given)

but it doesnt give `reverse-proxy-server-1`.lan IP.

Is this normal or is it a bug?

**Comment by ainar-g** [comment]
AGH in general does not currently follow CNAME chains.  There are some exceptions, such as the legacy rewrite mechanism, but they're inconsistent.  If you want to build a kind of your own DNS zone, it's better to use a list with `$dnsrewrite` rules for that.

**Reply by Buco7854** [reply]
Hello, thanks for the answer. I'll look into this. I assume it should be used in custom filtering rule?

**Reply by Buco7854** [reply]
Ok I just setup those 2 rules as a test

```
reverse-proxy-server-1.lan$dnsrewrite=NOERROR;A;192.168.1.61
||mydomain.com^$dnsrewrite=NOERROR;CNAME;reverse-proxy-server-1.lan
```

however when using nslookup it cannot resolve mydomain.com I assume its because it resolving the CNAME but doesnt take its own record in account? Anyway to make it so it resolves "correctly"?
